datasource/hcp-packer-image/data.go

```diff
@@ -171,9 +147,29 @@ func (d *Datasource) Execute() (cty.Value, error) {
 	log.Printf("[INFO] Reading info from HCP Packer registry (%s) [project_id=%s, organization_id=%s, iteration_id=%s]",
 		d.config.Bucket, cli.ProjectID, cli.OrganizationID, d.config.IterationID)
 
-	iteration, err := d.getIteration(ctx, cli)
-	if err != nil {
-		return cty.NullVal(cty.EmptyObject), err
+	var iteration *models.HashicorpCloudPackerIteration
+	var channelID string
```

Comment:
Should these be mutually exclusive? 

From looking at the code it appears that a user can provide both values, with the iteration id taking precedence. It can be a little confusing to the user if they provide both but only the iteration id is being used.

Reply:
The mutual exclusivity is checked in `Configure`?

Reply:
Ah mi culpa that was a previous change 🤦